File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        ListNode temp = head;
        ArrayList<Integer> l1 = new ArrayList<Integer>();
        ArrayList<Integer> l2 = new ArrayList<Integer>();
        int j=0;
        while(temp!=null){
            if((j)%2==0){
                l1.add(temp.val);
            }
            if((j)%2!=0){
                l2.add(temp.val);
            }
            temp=temp.next;
            j++;
        }temp=head;
        for(int i=0;i<l1.size();i++){
            temp.val=l1.get(i);
            temp=temp.next;
        }
        for(int i=0;i<l2.size();i++){
            temp.val=l2.get(i);
            temp=temp.next;
        }
        return head;
    }
}

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        ListNode* odd = new ListNode(-1);
        ListNode* even = new ListNode(-1);
        ListNode* oddHead = odd;
        ListNode* evenHead = even;
        
        ListNode* iterator = head;
        
        int num = 1;
        
        while(iterator != NULL){
            ListNode* toDel = iterator;
            if((num++) & 1){
                odd->next = new ListNode(iterator->val);
                odd = odd->next;
            }
            else{
                even->next = new ListNode(iterator->val);
                even = even->next;
            }
            iterator = iterator->next;
            delete toDel;
        }
        odd->next = evenHead->next;
        odd = oddHead->next;
        delete oddHead;
        
        return odd;
    }
};